### ISSUE TITLE
Rule Change: add 'empty-case' as a STATEMENT_TYPE

### ISSUE THREAD

**Issue by aparajita** [eslint/eslint#15472]
### What rule do you want to change?

padding-line-between-statements

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
switch (status) {
  case 200:
  case 201:
    res.status(201).json(body)
    break

  case 400:
    res.status(400).json(null)
    break
}
```

### What does the rule currently do for this code?

With the options set to this, with the intention of requiring a blank line between _non-empty_ `case` clauses:

```js
  { blankLine: 'always', prev: 'case', next: 'case' }
```

I get a warning on the `case 201:` line.

### What will the rule do after it's changed?

No warnings.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

I'm proposing the addition of an additional STATEMENT_TYPE 'empty-case', which would apply to `SwitchCase` nodes where `consequent.length === 0`. So the options would look like this to eliminate warnings for the sample code:

```js
  { blankLine: 'always', prev: 'case', next: 'case' },
  { blankLine: 'never', prev: 'empty-case', next: ['case', 'empty-case'] }
```

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by aparajita** [comment]
Thanks, I tried copying it into a custom rule but couldn't get it to work. I'll try again.
